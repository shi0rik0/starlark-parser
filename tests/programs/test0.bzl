a = [[1, 2], [3, 4], [5, 6]]
b = [[1, 3, 5, 7], [2, 4, 6, 8]]
n = len(a)
p = len(a[0])
if p != len(b):
    exit(0)
m = len(b[0])
c = [[0]*m for _ in range(n)]

for k in range(p):
    for i in range(n):
        for j in range(m):
            c[i][j] += a[i][k] * b[k][j] 
print(c)