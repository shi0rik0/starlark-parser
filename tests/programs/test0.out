[Assign] IDENTIFIER(a)=LIST(LIST(INT(1),INT(2)),LIST(INT(3),INT(4)),LIST(INT(5),INT(6)))
[Assign] IDENTIFIER(b)=LIST(LIST(INT(1),INT(3),INT(5),INT(7)),LIST(INT(2),INT(4),INT(6),INT(8)))
[Assign] IDENTIFIER(n)=CALL(IDENTIFIER(len),IDENTIFIER(a))
[Assign] IDENTIFIER(p)=CALL(IDENTIFIER(len),SLICE(IDENTIFIER(a),Slice(INT(0)))
[If] 
NE(IDENTIFIER(p),CALL(IDENTIFIER(len),IDENTIFIER(b)))->{
    [Expr] CALL(IDENTIFIER(exit),INT(0))
}
[Assign] IDENTIFIER(m)=CALL(IDENTIFIER(len),SLICE(IDENTIFIER(b),Slice(INT(0)))
[Assign] IDENTIFIER(c)=LIST_COMPREHENSION(MUL(LIST(INT(0)),IDENTIFIER(m)),FOR(IDENTIFIER(_),CALL(IDENTIFIER(range),IDENTIFIER(n))))
[For] IDENTIFIER(k)=CALL(IDENTIFIER(range),IDENTIFIER(p)){
    [For] IDENTIFIER(i)=CALL(IDENTIFIER(range),IDENTIFIER(n)){
        [For] IDENTIFIER(j)=CALL(IDENTIFIER(range),IDENTIFIER(m)){
            [Assign] SLICE(SLICE(IDENTIFIER(c),Slice(IDENTIFIER(i)),Slice(IDENTIFIER(j))+=MUL(SLICE(SLICE(IDENTIFIER(a),Slice(IDENTIFIER(i)),Slice(IDENTIFIER(k)),SLICE(SLICE(IDENTIFIER(b),Slice(IDENTIFIER(k)),Slice(IDENTIFIER(j)))
        }
    }
}
[Expr] CALL(IDENTIFIER(print),IDENTIFIER(c))
